Purchases/Public/RCPurchases.m

```diff
@@ -519,83 +616,55 @@ - (void)restoreTransactionsForAppStoreAccount
                                         NSError * _Nullable error) {
                                [self dispatch:^{
                                    if (error) {
-                                       [self.delegate purchases:self failedToRestoreTransactionsWithError:error];
+                                       CALL_AND_DISPATCH_IF_SET(completion, nil, error);
                                    } else if (info) {
-                                       [self cachePurchaserInfo:info];
-                                       [self.delegate purchases:self restoredTransactionsWithPurchaserInfo:info];
+                                       [self cachePurchaserInfo:info forAppUserID:self.appUserID];
+                                       CALL_AND_DISPATCH_IF_SET(completion, info, nil);
                                    }
                                }];
                            }];
     }];
 }
 
-- (void)updateOriginalApplicationVersion
-{
-    [self.backend getSubscriberDataWithAppUserID:self.appUserID completion:^(RCPurchaserInfo * _Nullable info,
-                                                                             NSError * _Nullable error) {
-        if (error) {
-            [self dispatch:^{
-                [self.delegate purchases:self failedToUpdatePurchaserInfoWithError:error];
-            }];
-        } else if (info.originalApplicationVersion) {
-            [self dispatch:^{
-                [self.delegate purchases:self receivedUpdatedPurchaserInfo:info];
-            }];
-        } else {
-            [self receiptData:^(NSData * _Nonnull data) {
-                [self.backend postReceiptData:data
-                                    appUserID:self.appUserID
-                                    isRestore:NO
-                            productIdentifier:nil
-                                        price:nil
-                                  paymentMode:RCPaymentModeNone
-                            introductoryPrice:nil
-                                 currencyCode:nil
-                                   completion:^(RCPurchaserInfo * _Nullable info, NSError * _Nullable error) {
-                                       [self handleUpdatedPurchaserInfo:info error:error];
-                                   }];
-            }];
-        }
-    }];
+- (void)clearCaches {
+    if (self.appUserID) {
```

Comment:
Is it possible this is nil?

Reply:
I don't think so but string by appending string within will crash if I don't have it... 

Someone could pass one in I suppose but that's an invalid state...